### ISSUE TITLE
Double clicking on stage from two far apart coordinates trigger events

### ISSUE THREAD

**Issue by khairulashraff** [konvajs/konva#1162]
This is an extension to the issue [here](https://github.com/konvajs/konva/issues/252).

That issue fixes when two separate shapes are quickly clicked, it will not trigger a double click. But when one of them is the stage, double click still fires.

There's also another related issue though it may require some feedback first. If consecutive clicks are done on only the stage but very far apart (say several hundred pixels away), dblclick will still trigger. IMHO, this should not happen. If multiple clicks occurred several hundred pixels away, it is usually unintentional. An actual double click which will expect an action to occur would have be done in close proximity. I can't say the right number but around radius of 10?

Example:
https://jsbin.com/cevajucewi/edit?html,js,console,output

**Comment by lavrton** [comment]
> That issue fixes when two separate shapes are quickly clicked, it will not trigger a double click. But when one of them is the stage, double click still fires.

I wasn't able to reproduce that with the last Konva version.

> If consecutive clicks are done on only the stage but very far apart (say several hundred pixels away), dblclick will still trigger.

You are right. And looks like DOM events don't work that way. We may fix it. But I do see why we need it. I never saw issues with the current behaviour.

**Comment by khairulashraff** [comment]
> I wasn't able to reproduce that with the last Konva version.

You're right, I actually noticed that on the example on JSBin rather than my own app. I changed to use the latest version and it does not happen. I guess the template was using an outdated version of Konva.

> You are right. And looks like DOM events don't work that way. We may fix it. But I do see why we need it. I never saw issues with the current behaviour.

I understand. I think this can only come up for a rather specific use-case. As for mine, I'm doing something like a graph drawing tool where double clicking on an empty space will add a new node. I bound it to the stage and it kept adding new node as I work on it.
